packages/neo-one-client-core/src/user/RemoteUserAccountProvider.ts

```diff
@@ -0,0 +1,273 @@
+import {
+  Account,
+  AddressString,
+  Block,
+  ClaimTransaction,
+  Hash256String,
+  InvocationTransaction,
+  InvokeSendUnsafeReceiveTransactionOptions,
+  IterOptions,
+  NetworkType,
+  Param,
+  RawAction,
+  RawCallReceipt,
+  RawInvokeReceipt,
+  ScriptBuilderParam,
+  SourceMaps,
+  TransactionOptions,
+  TransactionReceipt,
+  TransactionResult,
+  Transfer,
+  UpdateAccountNameOptions,
+  UserAccount,
+  UserAccountID,
+  UserAccountProvider,
+} from '@neo-one/client-common';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { Endpoint, Message } from './messageTypes';
+
+// tslint:disable no-let
+let MESSAGE_ID = 0;
+const getID = () => {
+  const currentID = MESSAGE_ID;
+  MESSAGE_ID += 1;
+
+  return `${currentID}`;
+};
+
+async function* asyncGenerator(messageEndpoint: Endpoint, method: string, network: NetworkType, options?: IterOptions) {
+  const id = getID();
+  messageEndpoint.postMessage({
+    type: 'ASYNCITERABLE',
+    method,
+    args: [network, options],
+    id,
+  });
+  // tslint:disable-next-line:no-loop-statement
+  while (true) {
+    // tslint:disable-next-line no-any
+    const rawActionIterator = new Promise<any>((resolve, reject) => {
+      const listener = (event: Message) => {
```

Comment:
This bit is identical to the `handleRelayMethod` logic, can you pull it out into a common function?